File: testdata/vulnerable.kt
import java.security.MessageDigest
import javax.crypto.Cipher
import java.security.KeyPairGenerator

class Vulnerable {
    fun badMD5(data: ByteArray): ByteArray {
        val md = MessageDigest.getInstance("MD5")
        return md.digest(data)
    }

    fun badSHA1(data: ByteArray): ByteArray {
        val md = MessageDigest.getInstance("SHA-1")
        return md.digest(data)
    }

    fun badDES(): Cipher = Cipher.getInstance("DES/ECB/PKCS5Padding")

    fun bad3DES(): Cipher = Cipher.getInstance("DESede/ECB/PKCS5Padding")

    fun weakRSA(): KeyPairGenerator {
        val kpg = KeyPairGenerator.getInstance("RSA")
        kpg.initialize(1024)
        return kpg
    }

    fun disallowedDSA(): KeyPairGenerator =
        KeyPairGenerator.getInstance("DSA")
}

File: testdata/vulnerable.scala
import java.security.MessageDigest
import javax.crypto.Cipher

object Vulnerable {
  def badMD5(data: Array[Byte]): Array[Byte] = {
    val md = MessageDigest.getInstance("MD5")
    md.digest(data)
  }

  def badSHA1(data: Array[Byte]): Array[Byte] = {
    val md = MessageDigest.getInstance("SHA-1")
    md.digest(data)
  }

  def badDES(): Cipher = Cipher.getInstance("DES/ECB/PKCS5Padding")

  def bad3DES(): Cipher = Cipher.getInstance("DESede/ECB/PKCS5Padding")
}
